### DISCUSSION TITLE
Getting bathymetry into Rayvista

### DISCUSSION THREAD

**Discussion by monocilindro** [h-a-graham/rayvista#25] [Ideas]
Would it be possible to automatically download and import bathymetry into Rayvista? GEBCO and EMODNET datasets might be easily imported for that purpose

**Comment by h-a-graham** [comment]
Hey, yes this is theoretically possible for sure. I'll have a think about how best to do this as currently all elevation data is requested via elevatr which is (as far as I'm aware) only terrestrial data.

**Reply by h-a-graham** [reply]
Actually it looks like 'srtm15plus' as the elevation src might work already. I'll try to have a play and post an example...

**Comment by monocilindro** [comment]
Just in case and regarding the GEBCO and EMODNET bathymetry datasets, not only the DEM can be accessed and downloaded, but also the tinted hill-shading layers , to be used instead of satellite imagery.

**Reply by h-a-graham** [reply]
Right, so if you update {elevatr} to the latest version on github:
`devtools::install_github("jhollist/elevatr")`
Then you can access the new 'srtm15plus' . Also, update {rayvista} as I fixed a bug to allow negative values. We could add more overlay options but for shading using rayshader is a great option - this example should give you some ideas:
```
library(rayvista)
library(rayshader)
![hawaii](https://user-images.githubusercontent.com/43027949/125211506-c8698200-e29e-11eb-8095-540335fbd1b6.png)

.lat <- 21.026059
.long <- -157.507622
buff <- 400000

hawaii <- plot_3d_vista(lat = .lat, long = .long, radius = buff,
                        elevation_detail = 14, overlay_detail = 10,
                        elevation_src = 'srtm15plus', overlay_alpha = 0.75,
                        show_vista = FALSE)

hawaiihadow = ray_shade(hawaii$dem_matrix, zscale = 75, lambert = FALSE, multicore = TRUE)
hawaiilamb = ambient_shade(hawaii$dem_matrix, zscale = 75, multicore = TRUE)

base <- hawaii$dem_matrix %>%
  height_shade(texture='white')%>%
  add_shadow(hawaiihadow, 0.5) %>%
  add_shadow(hawaiilamb, 0) %>%
  add_overlay(., hawaii$texture,rescale_original=TRUE)

base %>%  plot_3d(., hawaii$dem_matrix, zscale=100,
          windowsize = 1800, zoom=0.2, phi=20, theta=80, solidcolor = 'grey70',
          water = TRUE, waterdepth = -10, wateralpha = 0.1, watercolor = "#7CB0FF",
          waterlinecolor = "grey90", waterlinealpha = 0.5)

render_snapshot('exports/hawaii.png')
```

**Reply by h-a-graham** [reply]
![hawaii](https://user-images.githubusercontent.com/43027949/125211519-dcad7f00-e29e-11eb-9456-33d4de98d2a1.png)
